app/src/main/java/de/grobox/transportr/favorites/trips/FavoriteTripPopupMenu.java

```diff
@@ -50,12 +50,14 @@ public boolean onMenuItemClick(MenuItem item) {
 				//getMenu().findItem(R.id.action_add_shortcut).setVisible(trip.isFavorite());
 				return true;
 			case R.id.action_add_shortcut:
-
-				String widgetName = context.getString(R.string.widget_name_favorite);
-				widgetName = String.format(widgetName, String.valueOf(trip.getFrom()), String.valueOf(trip.getTo()));
-
-				addShortcut(context, getShortcutIntent(context, 0, trip.getFrom(), trip.getVia(), trip.getTo()),
-						widgetName, R.mipmap.ic_quickwork_widget);
+				addShortcut(context, getShortcutIntent(context,
+							trip.getUid(),
+							trip.getNetworkId(),
+							((StoredLocation) trip.getFrom()).getUid(),
+							(trip.getVia() == null) ? null : ((StoredLocation) trip.getVia()).getUid(),
+							((StoredLocation) trip.getTo()).getUid()),
```

Comment:
`trip.getUid()` should be sufficient to identify a trip and get the `FavoriteTripItem` from the database. No need for the other data here.